classad/_base_expression.py

```diff
@@ -0,0 +1,119 @@
+import pyparsing as pp
+
+from typing import Iterable, Any, TYPE_CHECKING, Union
+
+if TYPE_CHECKING:
+    from ._expression import ClassAd
+    from ._primitives import Undefined, Error, HTCBool
+
+
+class Expression:
+    _expression: "Expression"
+
+    def evaluate(
+        self, key: Iterable = None, my: "ClassAd" = None, target: "ClassAd" = None
```

Comment:
Applies to all ``= None`` defaults. Remember to import ``Optional``.

```suggestion
        self, key: Optional[Iterable] = None, my: "Optional[ClassAd]" = None, target: "Optional[ClassAd]" = None
```